projects/rocm-core/rdhc/rdhc.py

```diff
@@ -1348,6 +1347,96 @@ def test_check_multinode_cluster_readiness(self):
             self.logger.info(f" {success_msg}")
             return TestStatus.PASS.value, success_msg
 
+    def test_check_atomic_operations(self):
+        """Test if atomic operations are enabled for GPU devices"""
+        self.logger.info("--Checking atomic operations support for GPU devices...")
+
+        # Find AMD GPU devices using lspci
+        stdout, stderr, ret_code = run_command("lspci -d 1002: -nn | grep -Ei 'Display controller|Processing accelerators|VGA compatible controller'", shell=True)
+
+        if ret_code != 0 or not stdout.strip():
+            self.logger.error("!!! No AMD GPU devices found")
+            return TestStatus.FAIL.value, "No AMD GPU devices found to check atomic operations."
+
+        gpu_devices = stdout.strip().split('\n')
+        self.logger.info(f"----Found {len(gpu_devices)} AMD GPU device(s)")
+
+        atomic_ops_status = []
+        devices_with_atomics = 0
+        devices_without_atomics = 0
+        check_failed_devices = 0
+
+        for gpu_line in gpu_devices:
+            # Extract PCI address (e.g., "01:00.0")
+            pci_address = gpu_line.split()[0]
```

Comment:
We may currently know that this value is indexed.  Since we are dealing across processes, lets make sure this is indexed or use try catch for error free handling.